### DISCUSSION TITLE
Convert time only to datetime

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana-infinity-datasource#364] [General]
Hello

Given time only data

13:00:12

Can UQL help me tack on todays date in front of it?

Thanks!

**Comment by yesoreyeram** [comment]
That is bit tricky without knowing which day.  but there is a workaround [here](https://grafana-infinity-datasource.herokuapp.com/explore?orgId=1&left=%7B%22datasource%22:%22Infinity%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22type%22:%22uql%22,%22source%22:%22inline%22,%22format%22:%22table%22,%22url%22:%22https:%2F%2Fjsonplaceholder.typicode.com%2Fusers%22,%22url_options%22:%7B%22method%22:%22GET%22,%22data%22:%22%22%7D,%22root_selector%22:%22%22,%22columns%22:%5B%5D,%22filters%22:%5B%5D,%22global_query_id%22:%22%22,%22uql%22:%22parse-csv%5Cn%7C%20extend%20%5C%22curr_date%5C%22%3Dtostring(%27$%7B__from:date:YYYY-MM-DD%7D%27)%5Cn%7C%20extend%20%5C%22full_date%5C%22%3Dstrcat(%5C%22curr_date%5C%22,%27%20%27,%5C%22ts%5C%22)%5Cn%7C%20project%20%5C%22full_date%5C%22%3Dtodatetime(%5C%22full_date%5C%22),%20%5C%22val%5C%22%3Dtonumber(%5C%22val%5C%22)%22,%22data%22:%22ts,val%5Cn13:00:12,12%5Cn13:01:12,24%5Cn13:02:12,30%5Cn%22%7D%5D,%22range%22:%7B%22from%22:%221658318412000%22,%22to%22:%221658318532000%22%7D%7D) with the grafana global variable macros.

```csv
ts,val
13:00:12,12
13:01:12,24
13:02:12,30
```

```kusto
parse-csv
| extend "curr_date"=tostring('${__from:date:YYYY-MM-DD}')
| extend "full_date"=strcat("curr_date",' ',"ts")
| project "full_date"=todatetime("full_date"), "val"=tonumber("val")
```

I say this is tricky, because the timestamps may overlap between dates during midnight... `23:59` and `00:01` 🤔

**Reply by yosiasz** [reply]
This is perfect thanks
